Rtlfs devices should be more then 3 on all nodes

<!-- **Are you in the right place?**
1. For issues or feature requests, please create an issue in this repository.
2. For general technical and non-technical questions, we are happy to help you on our [Rook.io Slack](https://slack.rook.io/).
3. Did you already search the existing open issues for anything similar? -->

**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
Cluster creation failed with error: `edgefs-op-cluster: failed to create cluster in namespace rook-edgefs. Failed to create [rook-edgefs] EdgeFS cluster configuration. Error: Rtlfs devices should be more then 3 on all nodes summary`

**Expected behavior:**
If no devices specified in cluster.yaml then `autoRtlfs` mode should be selected and 4 folders should be created on each node
**How to reproduce it (minimal and precise):**
Deploy cluster w/o any devices selected
<!-- Please let us know any circumstances for reproduction of your bug. -->

**File(s) to submit**:

* Cluster CR (custom resource), typically called `cluster.yaml`, if necessary
* Operator's logs, if necessary
* Crashing pod(s) logs, if necessary

 To get logs, use `kubectl -n <namespace> logs <pod name>`
When pasting logs, always surround them with backticks or use the `insert code` button from the Github UI.
Read [Github documentation if you need help](https://help.github.com/en/articles/creating-and-highlighting-code-blocks).

**Environment**:
* OS (e.g. from /etc/os-release):
* Kernel (e.g. `uname -a`):
* Cloud provider or hardware configuration:
* Rook version (use `rook version` inside of a Rook Pod):
* Storage backend version (e.g. for ceph do `ceph -v`):
* Kubernetes version (use `kubectl version`):
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/master/ceph-toolbox.html)):

--- a/pkg/operator/edgefs/cluster/cluster_deployment_config.go
+++ b/pkg/operator/edgefs/cluster/cluster_deployment_config.go
@@ -311,7 +311,6 @@ func (c *cluster) validateDeploymentConfig(deploymentConfig edgefsv1.ClusterDepl
 		return err
 	}
 
-	deploymentNodesCount := len(deploymentConfig.DevConfig)
 	if deploymentConfig.TransportKey == edgefsv1.DeploymentRtkvs {
 		if len(c.Spec.Storage.Directories) == 0 {
 			return fmt.Errorf("RTKVS configuration error: storage.directory has to " +
@@ -322,9 +321,12 @@ func (c *cluster) validateDeploymentConfig(deploymentConfig edgefsv1.ClusterDepl
 				" A per-node NVME KVSSD disks list is expected")
 		}
 	} else if deploymentConfig.TransportKey == edgefsv1.DeploymentRtlfs {
-		// Check directories devices count on all nodes
-		if len(c.Spec.Storage.Directories)*deploymentNodesCount < 3 {
-			return fmt.Errorf("Rtlfs devices should be more then 3 on all nodes summary")
+		// Check directories devices count on all nodes for autoRtlfs mode
+		deploymentNodesCount := len(deploymentConfig.DevConfig)
+		if len(c.Spec.Storage.Directories) > 0 {
+			if len(c.Spec.Storage.Directories)*deploymentNodesCount < 3 {
+				return fmt.Errorf("Rtlfs devices should be more then 3 on all nodes summary")
+			}
 		}
 	} else if deploymentConfig.TransportKey == edgefsv1.DeploymentRtrd {
 		// Check all deployment nodes has available disk devices

